### PR TITLE
Split dependencies

### PR DESCRIPTION
**Pull request by ghickman**
I can't see any prod configuration in the repo, so the only change we'd need to make to deploy this is `s/requirements.txt/requirements.prod.txt/`, and then we'd be installing several fewer packages.

Locally, a user can now run `make compile` to get updated requirements files. I'll look at scripting this further in the future so the compiled files and when to update them become an implementation detail, but for now, this moves us along nicely!

Fix: #198

### CODE REVIEW

**Review by meshy — APPROVED** [review]

File: README.markdown [hunk 1]
```diff
@@ -44,11 +44,11 @@ First you should install some OS libraries required for some packages, this can
 
     sudo apt-get install python3-dev libmemcached-dev zlib1g-dev libpq-dev
```
[the remark this answers is not in the document]
Comment (ghickman) [reply README.markdown:45]: Agreed, I want to rewrite most of it tbh, which is why I left a lot of this.  Captured in #239

File: README.markdown [hunk 2]
```diff
@@ -44,11 +44,11 @@ First you should install some OS libraries required for some packages, this can
...
 
-After this, install as you normally would a Django site (requirements.txt provided).
+After this, install as you normally would a Django site (requirements files are provided).
 
 e.g. (inside your virtualenv or whatever)
 
-    pip install -r requirements.txt
+    [uv] pip install -r requirements.prod.txt -r requirements.dev.txt
 
 Prepare the database (assuming you've got required database)
```
[the remark this answers is not in the document]
Comment (ghickman) [reply README.markdown:53]: captured in #239

File: README.markdown [hunk 3]
```diff
@@ -44,11 +44,11 @@ First you should install some OS libraries required for some packages, this can
 
     sudo apt-get install python3-dev libmemcached-dev zlib1g-dev libpq-dev
 
-After this, install as you normally would a Django site (requirements.txt provided).
+After this, install as you normally would a Django site (requirements files are provided).
 
 e.g. (inside your virtualenv or whatever)
 
-    pip install -r requirements.txt
+    [uv] pip install -r requirements.prod.txt -r requirements.dev.txt
```
[the remark this answers is not in the document]
Comment (ghickman) [reply README.markdown:51]: Yes, I think there's a lot we can do here with a scripted stack to remove the need for most of this type of documentation and also use the same stack in CI.  For example, `[just|make] test` would run the pytest installed in the correct venv in a way that a user could pass options to it locally.

[the remark this answers is not in the document]
Comment (meshy) [reply README.markdown:51]: Cool OK -- I'll leave that for the readme fix!

**Pull request merged**